clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -3218,6 +3218,21 @@ static bool ParseHeaderSearchArgs(HeaderSearchOptions &Opts, ArgList &Args,
   bool IsIndexHeaderMap = false;
   bool IsSysrootSpecified =
       Args.hasArg(OPT__sysroot_EQ) || Args.hasArg(OPT_isysroot);
+
+  // Expand a leading `=` to the sysroot if one was passed (and it's not a
+  // framework flag).
+  auto ConvertHeaderPath = [IsSysrootSpecified,
+                            &Opts](const llvm::opt::Arg *A,
+                                   bool IsFramework = false) -> std::string {
+    if (IsSysrootSpecified && !IsFramework && A->getValue()[0] == '=') {
```

Comment:
Might be nice to take `const llvm::opt::Arg &` instead. It also feels unsafe as we do not know if the argument has a value before you get it - you should do a `A->hasValue()` or assert it perhaps?

Reply:
Maybe? The thing coming from `Args.filtered` should be populated though? Even if it isn't, we shouldn't be entering the `for`-loop though. Perhaps `Args.filtered` should result in iterator that emits a const reference? That seems like a bigger, unrelated change though.

Added an assert and a test case. The arg parser will error out earlier if nothing is passed to `-I`; `argument to '-I' is missing`. Is that sufficient?

Reply:
> I think that Prefix or Expand would both be better than Convert. We aren't really converting this.

There, changed to `PrefixHeaderPath`.